src/beman/exemplar/CMakeLists.txt

```
@@ -13,13 +13,10 @@ target_sources(
         FILES ${PROJECT_SOURCE_DIR}/include/beman/exemplar/identity.hpp
 )
 
-set_target_properties(beman.exemplar PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON)
+target_link_libraries(beman.exemplar PUBLIC beman.exemplar.headers)
 
-install(
-    TARGETS beman.exemplar
-    EXPORT beman.exemplar
-    DESTINATION
-    $<$<CONFIG:Debug>:debug/>${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION $<$<CONFIG:Debug>:debug/>${CMAKE_INSTALL_BINDIR}
-    FILE_SET HEADERS DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+beman_install_targets(
+    TARGETS beman.exemplar.headers beman.exemplar
 )
+
+beman_install_export()
```

```suggestion
beman_install_targets(TARGETS beman.exemplar)
```

I've had a mind to introduce something like this myself, for what it's worth. I think it's a design gap in CMake upstream that there's not a clear design seam here.

That being said, I've had good luck elsewhere leaving it at only installing the library target and letting its properties and such speak for themselves. For instance the `FILE_SET HEADERS` properties document that it has headers, that they are installed, where they are installed, and so on.

I would also expect that installing the `beman.exemplar` module would be unconditional. It's phrased that way here since there's no `if` statement! Similarly, we'll (in another PR) support `pkg-config` and [CPS](https://cps-org.github.io/cps/overview.html) files in the future.

---

I should have included a comment that I meant for this to be demonstrative, not prescriptive. I didn't mean to imply that we should have a separate `headers` target for each library. I was just showing how to use the `beman_add_header_only_library()`. I do think it's worth having a separate function for that case to make it easier for library authors to do the "right" thing.

That said, I do find it frustrating when I have to link to a compiled target in order to get the proper include paths when all I need is a function template. When I publish libraries, I tend to include a headers target for that reason, but I don't think it's a commonly accepted practice.

---

Ah, makes sense. Though I'm not particularly interested in header-only as a use case. It doesn't really buy individual projects much. We still need packaging, CMake exports, unit tests, etc., so having a few cpp files or not doesn't save anyone anything, and it does add complexity when we're already over our complexity budget in CMakeLists.txt (IMO).